filesystem/File.php

```diff
@@ -224,28 +282,27 @@ public static function link_shortcode_handler($arguments, $content = null, $pars
 	/**
 	 * A file only exists if the file_exists() and is in the DB as a record
 	 *
+	 * Use $file->isInDB() to only check for a DB record
+	 * Use $file->File->exists() to only check if the asset exists
+	 *
 	 * @return bool
 	 */
 	public function exists() {
-		return parent::exists() && file_exists($this->getFullPath());
+		return parent::exists() && $this->File->exists();
 	}
 
 	/**
 	 * Find a File object by the given filename.
 	 *
-	 * @param String $filename Matched against the "Name" property.
-	 * @return mixed null if not found, File object of found file
+	 * @param string $filename Filename to search for, including any custom parent directories.
+	 * @return File
 	 */
 	public static function find($filename) {
-		// Get the base file if $filename points to a resampled file
-		$filename = Image::strip_resampled_prefix($filename);
-
 		// Split to folders and the actual filename, and traverse the structure.
 		$parts = explode("/", $filename);
 		$parentID = 0;
 		$item = null;
 		foreach($parts as $part) {
-			if($part == ASSETS_DIR && !$parentID) continue;
```

Comment:
Skipping `assets/` on the root level might still be a useful feature? It can lead to subtle bugs in existing code to drop this (only occurs for some files). You couldn't find a file with `<webroot>/assets/assets/somefile.png` with the check in place, but that's already a bug in the implementation. If it's removed, we have to be clearer about the expected file path in `@param`